framework/include/base/Attributes.h

```diff
@@ -247,6 +247,27 @@ class AttribThread : public Attribute
   THREAD_ID _val = 0;
 };
 
+class AttribExecutionOrderGroup : public Attribute
+{
+public:
+  typedef int Key;
+  void setFrom(Key k) { _val = k; }
+
+  AttribExecutionOrderGroup(TheWarehouse & w) : Attribute(w, "execution_order_group") {}
+  AttribExecutionOrderGroup(TheWarehouse & w, Key p)
+    : Attribute(w, "execution_order_group"), _val(p)
+  {
+  }
+  virtual void initFrom(const MooseObject * obj) override;
+  virtual bool isMatch(const Attribute & other) const override;
+  virtual bool isEqual(const Attribute & other) const override;
+  hashfunc(_val);
+  clonefunc(AttribExecutionOrderGroup);
+
+private:
+  THREAD_ID _val = 0;
```

Comment:
```suggestion
  int _val = 0;
```

Reply:
Actually... this might allow you to not run with negative values. THREAD_ID is uint.

@GiudGiud said you were testing negatives but I can't seem to see it?